reconfigure.bash

```diff
@@ -23,6 +23,16 @@ elif [[ $# -eq 0 ]] && [[ ! -f "${script_dir}/role" ]]; then
   usage
 fi
 
+# Confirm that node-specific facts are usable.
+if ! grep -q '^trusted_node_data ?= ?true' /etc/puppet/puppet.conf; then
```

Comment:
I think that the check for optional space before and after the = is not working (if you run twice the whole bash code you end up with the command inserted twice). This should work the grep pattern:
 
```suggestion
if ! grep -q ''^trusted_node_data[[:space:]]*=[[:space:]]*true'' /etc/puppet/puppet.conf; then
``

Reply:
Is there some power to doubled single-quotes that I'm not aware of? Or is that part of the suggestion a typo?

Reply:
I think this also may just need the Extended Regular Expression engine.
```suggestion
if ! grep -Eq '^trusted_node_data ?= ?true' /etc/puppet/puppet.conf; then
```

Would you give that a try @j-rivero?